Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go

```diff
@@ -2865,6 +2870,17 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 	return nil
 }
 
+func (kl *Kubelet) updateCloudProviderFromMachineInfo(
+	node *api.Node,
+	info *cadvisorapi.MachineInfo) {
+
+	if info.CloudProvider != cadvisorapi.UnknownProvider &&
+		info.CloudProvider != cadvisorapi.Baremetal {
+		node.Spec.ProviderID = strings.ToLower(string(info.CloudProvider)) +
+			":////" + string(info.InstanceID)
```

Comment:
Usually it returns <cloud-provider>://<project_name>/<avilability_zone>/<instance_id> . we dont have the project name or the availability zone so they are left empty (kubelet's cloud providers may also return them empty when they are not available)